datasetCreation.py

```diff
@@ -53,18 +66,105 @@ def __init__(self):
         self.layer_output = nn.Sequential(
             nn.Linear(1024, 512), nn.BatchNorm2d(256), nn.Sigmoid())
 
-    def forward(x,self):
-       x = self.layer_input(x)
-       x = self.layer1(x)
-       x = self.layer2(x)
-       x = self.layer3(x)
-       x = self.layer4(x)
-       return (self.layer_output(x))
+    def forward(self, x):
+        x = self.layer_input(x)
+        x = self.layer1(x)
+        x = self.layer2(x)
+        x = self.layer3(x)
+        x = self.layer4(x)
+        return self.layer_output(x)
+
+
+def collate_fn_pad(batch):
+    """
+    Takes images as tensors and labels as input, finds highest and widest size of an image than pad smaller images.
+    :param batch: list of images and labels [img_as_tensor, 'label', ....]
+    :return: returns tuple where ([padded_img_as_tensors,..], ('label',..))
+    :rtype: (list, tuple)
+    """
+    imgs, labels = zip(*batch)
+    # get max width and height
+    h, w = zip(*[list(t[0].size()) for t in imgs])
+    max_h, max_w = max(h), max(w)
+
+    padded_imgs = []
+    # padding
+    for img in imgs:
+        pad_h = max_h - img[0].size(0)
+        pad_w = max_w - img[0].size(1)
+
+        pad_l = int(pad_w/2)  # left
+        pad_r = pad_w - pad_l  # right
+        pad_t = int(pad_h/2)  # top
+        pad_b = pad_h - pad_t  # bottom
+        pad = nn.ZeroPad2d((pad_l, pad_r, pad_t, pad_b))
+        padded_imgs.append(pad(img[0]))
+
+    return padded_imgs, labels
+
+
+def split_dataset(train_data, test_data, batch_size=1, shuffle=True, random_seed=42):
+    """
+    Splits dataset to training and testing set by given ratio.
+    :param test_data: dataset with testing data from DataCreator
+    :param train_data: dataset with training data from DataCreator
+    :param batch_size: size of the batch
+    :param random_seed: seed for random shuffling
+    :param shuffle: bool - decides if dataset will be shuffled
+    :return:
+    :rtype: training and testing dataset
+    """
+
+    train_data_size = len(train_data)
+    test_data_size = len(test_data)
+    train_indices, test_indices = list(range(train_data_size)), list(range(test_data_size))
+    if shuffle:
+        np.random.seed(random_seed)
+        np.random.shuffle(test_indices)
+        np.random.shuffle(train_indices)
+
+    train_sampler = SubsetRandomSampler(train_indices)
+    test_sampler = SubsetRandomSampler(test_indices)
+
+    train_loader = DataLoader(train_data, batch_size=batch_size, sampler=train_sampler,
+                              collate_fn=collate_fn_pad, drop_last=True)
+    test_loader = DataLoader(test_data, batch_size=batch_size, sampler=test_sampler,
+                             collate_fn=collate_fn_pad, drop_last=True)
+
+    return train_loader, test_loader
+
+
+def test_data_loaders(train_data, test_data):
```

Comment:
očividně se jedná jen o debug funkci, nehodilo by se vytvořit wrapper kde by se na základě bool hodnoty (DEBUG) spouštěla daná funkce?. Proměnou DEBUG bych přidal jako argument s defaultní hodnotou false

Reply:
zdroje k dekorátoru:

https://pythonbasics.org/decorators/
https://www.programiz.com/python-programming/decorator